src/GraphQL.MemoryCache/MemoryCacheGraphQLBuilderExtensions.cs

```diff
@@ -23,19 +23,27 @@ public static IGraphQLBuilder AddMemoryCache(this IGraphQLBuilder builder, Actio
         return builder.AddDocumentCache<MemoryDocumentCache>();
     }
 
+    /// <inheritdoc cref="UseAutomaticPersistedQueries(IGraphQLBuilder, Action{AutomaticPersistedQueriesCacheOptions}?)"/>
+    [Obsolete("Please use UseAutomaticPersistedQueries. This method will be removed in v8.")]
+    public static IGraphQLBuilder AddAutomaticPersistedQueries(this IGraphQLBuilder builder, Action<AutomaticPersistedQueriesCacheOptions>? action = null)
+        => UseAutomaticPersistedQueries(builder, action);
+
+    /// <inheritdoc cref="UseAutomaticPersistedQueries(IGraphQLBuilder, Action{AutomaticPersistedQueriesCacheOptions, IServiceProvider}?)"/>
+    [Obsolete("Please use UseAutomaticPersistedQueries. This method will be removed in v8.")]
+    public static IGraphQLBuilder AddAutomaticPersistedQueries(this IGraphQLBuilder builder, Action<AutomaticPersistedQueriesCacheOptions, IServiceProvider>? action)
+        => UseAutomaticPersistedQueries(builder, action);
+
     /// <summary>
     /// Adds support of Automatic Persisted Queries in the form of implementation of <see cref="IConfigureExecution"/>.
     /// https://www.apollographql.com/docs/react/api/link/persisted-queries/
     /// </summary>
-    public static IGraphQLBuilder AddAutomaticPersistedQueries(this IGraphQLBuilder builder, Action<AutomaticPersistedQueriesCacheOptions>? action = null)
-        => builder.AddAutomaticPersistedQueries(action == null ? null : (options, _) => action(options));
+    public static IGraphQLBuilder UseAutomaticPersistedQueries(this IGraphQLBuilder builder, Action<AutomaticPersistedQueriesCacheOptions>? action = null)
+        => builder.UseAutomaticPersistedQueries(action == null ? null : (options, _) => action(options));
 
-    /// <inheritdoc cref="AddAutomaticPersistedQueries(IGraphQLBuilder, Action{AutomaticPersistedQueriesCacheOptions})"/>
-    public static IGraphQLBuilder AddAutomaticPersistedQueries(this IGraphQLBuilder builder, Action<AutomaticPersistedQueriesCacheOptions, IServiceProvider>? action)
+    /// <inheritdoc cref="UseAutomaticPersistedQueries(IGraphQLBuilder, Action{AutomaticPersistedQueriesCacheOptions})"/>
+    public static IGraphQLBuilder UseAutomaticPersistedQueries(this IGraphQLBuilder builder, Action<AutomaticPersistedQueriesCacheOptions, IServiceProvider>? action)
     {
-        builder.Services
-            .Configure(action)
-            .TryRegister<IConfigureExecution, AutomaticPersistedQueriesExecution>(ServiceLifetime.Singleton, RegistrationCompareMode.ServiceTypeAndImplementationType);
```

Comment:
I see switching from TryRegister to Register semantics and I think it's fine in general case.

Reply:
If you remember, I suggested we do not implement `RegistrationCompareMode.ServiceTypeAndImplementationType` at all.  But since we have it, I was going to write a subsequent PR to change many of the `AddSomething<T>()` registrations where it would be logical to do so (not the `AddSomething<T>(T instance)` or `AddSomething<T>(factory)` calls).

As of now, this particular change here is made to be similar to the rest of the similar calls in the library -- but again, I was going to make another PR which would change this back.

Reply:
I was thinking about it like about `UseDeveloperExceptionPage`. If one wants to use it multiple times - why not ?